sidechain/consensus/aura/src/block_importer.rs

```diff
@@ -159,6 +179,61 @@ impl<
 	pub(crate) fn block_author_is_self(&self, block_author: &SignedSidechainBlock::Public) -> bool {
 		self.authority.public() == *block_author
 	}
+
+	fn create_finish_game_extrinsic(
+		&self,
+		sidechain_block: &&<SignedSidechainBlock as SignedBlock>::Block,
+	) -> Result<(), ConsensusError> {
+		let calls = self
+			.top_pool_executor
+			.get_trusted_calls(&sidechain_block.shard_id())
+			.map_err(|e| ConsensusError::Other(format!("{:?}", e).into()))?;
+		for call in calls {
+			if let TrustedCall::connectfour_play_turn(account, _b) = call.call {
+				let shard = &sidechain_block.shard_id();
+				let mut state = self
+					.state_handler
+					.load_initialized(shard)
+					.map_err(|e| ConsensusError::Other(format!("{:?}", e).into()))?;
+				let res = state.execute_with(|| get_board_for(account));
+				if let Some(board) = res {
+					if let BoardState::Finished(_) = board.board_state {
+						// player 1 is red, player 2 is blue
+						// the winner is not the next player
+						let winner = match board.next_player {
+							1 => board.blue.clone(),
+							_ => board.red.clone(),
+						};
+
+						let opaque_call = OpaqueCall::from_tuple(&(
+							[GAME_REGISTRY_MODULE, FINISH_GAME],
+							shard,
+							winner,
+						));
+
+						let calls = vec![opaque_call];
+
+						// Create extrinsic for acknowledge game.
+						let finish_game_extrinsic = self
+							.extrinsics_factory
+							.create_extrinsics(calls.as_slice())
+							.map_err(|e| ConsensusError::Other(format!("{:?}", e).into()))?;
+
+						// Sending the extrinsic requires mut access because the validator caches the sent extrinsics internally.
+						self.validator_accessor
+							.execute_mut_on_validator(|v| {
+								v.send_extrinsics(self.ocall_api.as_ref(), finish_game_extrinsic)
+							})
+							.map_err(|e| ConsensusError::Other(format!("{:?}", e).into()))?;
+						error!("sending extrinsic finisch game")
+					}
+				} else {
+					error!("could not decode board. maybe hasn't been set? {:x?}", res);
```

Comment:
```suggestion
					error!("could not decode board. maybe hasn't been set?");
```
If it's not Some(board), it will be None. So printing that will most likely not help that much.